server-spi-private/src/main/java/org/keycloak/userprofile/DefaultUserProfile.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *
+ *  * Copyright 2021  Red Hat, Inc. and/or its affiliates
+ *  * and other contributors as indicated by the @author tags.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.keycloak.userprofile;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.keycloak.models.UserModel;
+
+/**
+ * <p>The default implementation for {@link UserProfile}. Should be reused as much as possible by the different implementations
+ * of {@link UserProfileProvider}.
+ *
+ * <p>This implementation is not specific to any user profile implementation.
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class DefaultUserProfile implements UserProfile {
+
+    private final Function<Attributes, UserModel> userSupplier;
+    private final DefaultAttributes attributes;
+    private boolean validated;
+
+    public DefaultUserProfile(DefaultAttributes attributes, Function<Attributes, UserModel> userSupplier) {
```

Comment:
What is this userSupplier, how should it work? Should be documented here as generic Function is used.

Reply:
Internal thing, not part of public API.